docs/GUIDE.md

```diff
@@ -0,0 +1,142 @@
+
+- `org.virtuslab.ash.circe.compression.compress-larger-than` = `64 KiB` _(example value)_
+
+If `org.virtuslab.ash.circe.compression.algorithm` = `gzip` and `compress-larger-than` **value** is larger than `0 KiB` &mdash; each payload greater than the **value** will be compressed before serialization. Default **value** is `32 KiB`. If `org.virtuslab.ash.circe.compression.algorithm` = `off` &mdash; payloads will not be compressed, regardless of the chosen `compress-larger-than` value.<br><br>
+**Note** &mdash; Circe Akka Serializer will perform **deserialization** properly for both compressed and not compressed payloads regardless of both `org.virtuslab.ash.circe.compression` configurations (Serializer will recognize whether the payload has been compressed programmatically). These settings matter only for choosing type of serialization.
+
+### Annotations Guide
+ASH compiler plugins are based on usages of two annotations: [@SerializabilityTrait](#SerializabilityTrait) and [@Serializer](#Serializer). Thus, before enabling compiler plugins, make sure that you are using these two annotations properly in the project / module where plugins will do their work. In order to have access to these annotations, add this part of ASH to library dependencies:
+```sbt
+import org.virtuslab.ash.AkkaSerializationHelperPlugin
+(...)
+libraryDependencies += AkkaSerializationHelperPlugin.annotation
+```
+#### SerializabilityTrait
+[@SerializabilityTrait](../annotation/src/main/scala/org/virtuslab/ash/annotation/SerializabilityTrait.scala) is an annotation that should be added to your top-level serializable type (trait mentioned in the `akka.actor.serialization-bindings` part of akka config) that is extended by Messages / Events / States. Moreover, if your top-level serializable type is extended by another **trait** &mdash; this trait should also be annotated with this annotation. Concrete classes extending "serializability traits" (i.e. classes that define Messages/Events/States) should not be marked with this annotation. See examples in [README.md](../README.md)
+#### Serializer
+[@Serializer](../annotation/src/main/scala/org/virtuslab/ash/annotation/Serializer.scala) is an annotation used by the [Codec Registration Checker Compiler Plugin](#codec-registration-checker-compiler-plugin-guide) to check if there are any missing codec registrations. Add the `@Serializer` annotation to each serializer listed in the `akka.actor.serializers` part of the akka config. Moreover, to achieve full assurance that all codecs will be registered properly &mdash; add `@Serializer` annotation to each class/object/trait that contains code responsible for codec registration. See [Serializer Scaladoc](../annotation/src/main/scala/org/virtuslab/ash/annotation/Serializer.scala) for more details.
```

Comment:
`Akka` uppercase everywhere (when refering to software name from the **prose**, ofc not in code/configs etc.)